### PR TITLE
Better cifar10.py

### PR DESCRIPTION
**Pull request by ahundt**
cifar10.py provides more debug output, performance charts, configurable batch sizes, supports TensorBoard, and provides better information about how the run is going, and to put checkpoints in folders based on time so they aren't overwritten inadvertently.

### CODE REVIEW

File: cifar10.py [hunk 1]
```diff
@@ -29,58 +43,147 @@
...
+        # (std, mean, and principal components if ZCA whitening is applied).
+        datagen.fit(X_train)
+        if not os.path.exists(out_dir):
+            os.makedirs(out_dir)
+        
+        csv = CSVLogger(out_dir+dirname+'.csv', separator=',', append=True)
+        model_checkpoint = ModelCheckpoint(out_dir+'weights.hdf5', monitor='val_loss', verbose=0, save_best_only=True, save_weights_only=False, mode='auto')
+        callbacks=[lr_reducer, early_stopper, csv]
+        
+        if K.image_dim_ordering() == 'tf':
```
Comment (raghakot) [inline cifar10.py]: image dim ordering doesn't imply tensorflow backend. Is there a way to check 'backend' instead?

Comment (ahundt) [inline cifar10.py]: You're right, but I don't know the correct way... https://keras.io/backend/

Comment (raghakot) [inline cifar10.py]: Use K.backend() method. It is at the end of the doc you linked.
